WordPress/src/main/java/org/wordpress/android/ui/posts/reactnative/ReactNativeRequestHandler.kt

```diff
@@ -73,11 +76,35 @@ class ReactNativeRequestHandler @Inject constructor(
     private fun handleResponse(
         response: ReactNativeFetchResponse,
         onSuccess: (String) -> Unit,
-        onError: (String) -> Unit
+        onError: (Bundle) -> Unit
     ) {
         when (response) {
             is Success -> onSuccess(response.result.toString())
-            is Error -> onError(response.error)
+            is Error -> {
+                val bundle = Bundle().apply {
+                    response.error.volleyError?.networkResponse?.statusCode?.let {
+                        putInt("code", it)
+                    }
+                    extractErrorMessage(response.error)?.let {
+                        putString("message", it)
+                    }
+                }
+                onError(bundle)
```

Comment:
I don't know if it's possible but what about sending a `Pair` instead? It seems less error prone than a bundle 🤷‍♂

Reply:
`Pair` is actually what I used for my first implementation of this. 😀

I decided to use a `Bundle` instead because I had to convert the `Pair` to a `Bundle`-like React Native object (`WritableMap`) that can be sent across the javascript bridge to React Native (like with saving instance state for `Activities` you can only send a few special kinds of objects across the javascript bridge). That conversion involves exactly the same kind of `putString(...)` and `putInt(...)` calls that I am doing here, so it didn't seem like `Pair` was gaining me anything, it was just moving the place where I did the "risky" `putX(...)` calls. Using a `Bundle` directly seemed to make since on account of the fact that React Native has a helper method for converting a `Bundle` to a `WritableMap` automatically. 

There is a helper for converting `Map<String, Object` to `WritableMap` as well, but that actually seemed less safe since we would have no type safety to insure that we only put objects into the map that could be sent across the bridge.

Using `Bundle` directly also seemed like a better longterm approach since if/when (most likely when) we end up wanting to pass a third piece of information from the error, I think we would need to convert this back from a `Pair` to a `Bundle` or start nesting `Pair<Pair<>>`. 😱 (

An alternative approach would be to change `WordPress-Editor` to use `react-native-gutenberg-bridge` as an `api` dependency instead of an `implementation` dependency. That would allow us to define a well-typed object for the relevant error information in `react-native-gutenberg-bridge` and use that object here in `WordPress-Android` (transitively, via the Editor module). I feel like this might actually end up being the best longterm solution, but it adds React Native dependencies to the WordPress app itself (as opposed to just being implementation dependencies of the Editor module) and I don't feel comfortable that I've thought through all the implications of doing that yet. wdyt?